src/mbgl/gl/upload_pass.cpp

```diff
@@ -84,6 +84,26 @@ std::unique_ptr<gfx::TextureResource> UploadPass::createTextureResource(const Si
     return resource;
 }
 
+gfx::Texture2DPtr UploadPass::createTexture2D(const Size size,
+                                              const void* data,
+                                              gfx::TexturePixelType format,
+                                              gfx::TextureChannelDataType type) {
+    auto tex = std::make_shared<gl::Texture2D>(commandEncoder.context);
+    int textureByteSize = gl::TextureResource::getStorageSize(size, format, type);
+    commandEncoder.context.renderingStats().memTextures += textureByteSize;
+    commandEncoder.context.pixelStoreUnpack = {1};
+    tex->create(data, *this);
```

Comment:
```
tex->setFormat(format, type);
tex->create(data, *this);
```

Reply:
And `setSize`? That's probably why I was seeing zeros.

Reply:
Yeah, I've got a second pass in progress stacked on this PR which should make things a bit nicer. Can go ahead and merge this and follow it up on my second pass.

Reply:
Feel free to commit directly unless there's something you want to discuss first.
FYI, it there are some `Texture2D` merge conflicts with #1195.